core.go

```diff
@@ -412,6 +418,49 @@ func (m *Mat) SetDoubleAt3(x, y, z int, val float64) {
 	C.Mat_SetDouble3(m.p, C.int(x), C.int(y), C.int(z), C.double(val))
 }
 
+// ToImage returns Go image.Image from Mat.
+func (m *Mat) ToImage() (image.Image, error) {
+	t := m.Type()
+	if t != int(MatTypeCV8U) && t != int(MatTypeCV8UC1) && t != int(MatTypeCV8UC3) && t != int(MatTypeCV8UC4) {
+		return nil, errors.New("ToImage supports only MatType CV8U, CV8UC1, CV8UC3 and CV8UC4")
+	}
+
+	var height, width, channels, step int = m.Rows(), m.Cols(), m.Channels(), m.Step()
+
+	data := m.ToBytes()
+
+	if t == int(MatTypeCV8U) || t == int(MatTypeCV8UC1) {
```

Comment:
Could you tell me what these two types have in common?

I would add a method on Mat instead of doing this comparison here, so that we can reuse it.

Reply:
Actually, I am not sure if MatTypeCV8U is correct, I did a local test with some images I converted with imagemagick, so for now better to remove it.